RLP-1008 Updating readme and docs (#173)
anarancio/nifty-wallet: docs/configuration.md

              multiChainResolver: '',
              rif: '',
              fifsAddrRegistrar: '',
              rskOwner: '',
            },
          },
        };
    }
  }
```
**Note: We only show the important piece of code here.**

As you can see we have a structure like this:

```json5
{
  lumino: {
    explorer: {
      endpoint: '',
    },
  },
  notifier: {
    availableNodes: [],
  },
  rns: {
    contracts: {
      rns: '',
      publicResolver: '',
      multiChainResolver: '',
      rif: '',
      fifsAddrRegistrar: '',
      rskOwner: '',
    },
  },
}
```

Now we need to fill in those empty fields. Here are references about what those fields are:

```json5
{
  lumino: {
    explorer: {
      endpoint: 'endpoint URL for the Lumino Explorer API; e.g. http://localhost:8080/api/v1',
    },
  },
  notifier: {
    availableNodes: [
      // This is an array of notifier endpoints, you need at least 3.
      'Notifier 1 running API endpoint, e.g.: http://localhost:8081/',
      'Notifier 2 running API endpoint, e.g.: http://localhost:8082/',
      'Notifier 3 running API endpoint, e.g.: http://localhost:8083/'
    ],
  },
  rns: {
    contracts: {

>       // All these values can be retrieved from the RNS Suite deployment summary.

Author: ```suggestion
      // All these values can be retrieved from the RNS Suite (https://github.com/rnsdomains/rns-suite) deployment summary.
```

## How to configure RIF Services for local development.

> When you run RConnect on Regtest with a local node you need to setup all the 

Reviewer: ```suggestion
When you run RConnect on Regtest with a local node you need to set up all the 
```

neccesary infrastructure to allow the wallet to work with RIF Services. 

> To do so you need to configure some information about the API's that allow

Reviewer: ```suggestion
To do so you need to configure some API parameters that allow
```

us to access those services.

> If we see the RConnect General Architecture Diagram:

Reviewer: ```suggestion
Take a look at the RConnect General Architecture Diagram:
```

(basically the same, i know haha)

> ![General Diagram](./general-rconnect-architecture.png)

Reviewer: it makes sense to have this here, but what doesn't seem so logical to me is that:
1. we also have this diagram on the main `readme.md` file
2. we have an even _more_ specific diagram there as well

i would maybe just have both here and link them from there

There we have 1 RSK Node, 1 Lumino Explorer, 1 Lumino Hub and several Notifiers.

> All these are API's running somewhere, so we need to setup those to let the wallet
> to access them. 

Reviewer: ```suggestion
All these are running APIs, so we need to specify where they are in order to let the wallet access them.
```

> The file that let us configure this is [this one.](../app/scripts/controllers/rif/configuration/index.js)

Reviewer: ```suggestion
The file that lets us configure this is [`index.js`.](../app/scripts/controllers/rif/configuration/index.js)
```

> There we will see that is a JS file with some functions inside, basically what we need
> to change is the function `getInitialConfigStructure` that makes the initial setup.

Reviewer: ```suggestion
This is a JS file which defines a particular set of functions. We need to change the `getInitialConfigStructure` function because it is the one that does the initial setup.
```

This function looks similar to this:

```js
  getInitialConfigStructure (chainId) {
    // ... some other code
      case global.networks.reg: // RSK Regtest
        return {
          lumino: {
            explorer: {
              endpoint: '',
            },
          },
          notifier: {
            availableNodes: [],
          },
          rns: {
            contracts: {
              rns: '',
              publicResolver: '',
              multiChainResolver: '',
              rif: '',
              fifsAddrRegistrar: '',
              rskOwner: '',
            },
          },
        };
    }
  }
```
**Note: We only show the important piece of code here.**

As you can see we have a structure like this:

```json5
{
  lumino: {
    explorer: {
      endpoint: '',
    },
  },
  notifier: {
    availableNodes: [],
  },
  rns: {
    contracts: {
      rns: '',
      publicResolver: '',
      multiChainResolver: '',
      rif: '',
      fifsAddrRegistrar: '',
      rskOwner: '',
    },
  },
}
```

> Now we need to fill those empty fields, here is some reference about what
> those fields are:

Reviewer: ```suggestion
Now we need to fill in those empty fields. Here are references about what those fields are:
```

```json5
{
  lumino: {
    explorer: {

>       endpoint: 'Here you need to put the endpoint url where the Lumino Explorer API is running, for example http://localhost:8080/api/v1',

Reviewer: ```suggestion
      endpoint: 'endpoint URL for the Lumino Explorer API; e.g. http://localhost:8080/api/v1',
```

    },
  },
  notifier: {
    availableNodes: [
      // This is an array of notifier endpoints, you need at least 3.

>       'The endpoint for the notifier 1 (where the API is running), for example: http://localhost:8081/',
>       'The endpoint for the notifier 2 (where the API is running), for example: http://localhost:8082/',
>       'The endpoint for the notifier 3 (where the API is running), for example: http://localhost:8083/'

Reviewer: ```suggestion
      'Notifier 1 running API endpoint, e.g.: http://localhost:8081/',
      'Notifier 2 running API endpoint, e.g.: http://localhost:8082/',
      'Notifier 3 running API endpoint, e.g.: http://localhost:8083/'
```

    ],
  },
  rns: {
    contracts: {

>       // All these values can be retrieved from the RNS Suite deployment summary.

Reviewer: link here would be useful

      rns: 'The RNS Contract Address',
      publicResolver: 'The Public Resolver Contract Address',
      multiChainResolver: 'The Multi-chain Resolver Contract Address',
      rif: 'The RIF Token Address',
      fifsAddrRegistrar: 'The FIFSAddrRegistrar Contract Address',
      rskOwner: 'The RSK Owner Contract Address',
    },
  },
}
```

> As you can see we need to setup some endpoints (for Lumino Explorer and Notifiers) and then the RNS Suite Contract Addresses.

Reviewer: ```suggestion
As you can see we need to set up some endpoints (for Lumino Explorer and RIF Notifiers) and then RNS Suite Smart Contract addresses.
```

> Lumino Hub doesn't need to be here since Lumino Explorer provide that for us in runtime so we are ok.

Reviewer: ```suggestion
Lumino Hub data doesn't need to be included here since the Lumino Explorer provides that for us in runtime, so we're OK in that regard.
```

> Here is an example of a local configuration:

Reviewer: awesome, very much needed :+1:

```json5
{
  lumino: {
    explorer: {
      endpoint: 'http://localhost:8080/api/v1',
    },
  },
  notifier: {
    availableNodes: [
      'http://localhost:8081/',
      'http://localhost:8082/',
      'http://localhost:8083/',
    ],
  },
  rns: {
    contracts: {
      rns: '0x83C5541A6c8D2dBAD642f385d8d06Ca9B6C731ee',
      publicResolver: '0xE0825f57Dd05Ef62FF731c27222A86E104CC4Cad',
      multiChainResolver: '0x73ec81da0C72DD112e06c09A6ec03B5544d26F05',
      rif: '0x1Af2844A588759D0DE58abD568ADD96BB8B3B6D8',
      fifsAddrRegistrar: '0x8921BF2f074b5470c02Cc7473F17282576111591',
      rskOwner: '0x79bbC6403708C6578B0896bF1d1a91D2BB2AAa1c',
    },
  },
}
```

**IMPORTANT: These parameters are only initial parameters, the wallet needs to be

> re-installed (removed and installed again) on the browser after we change this, otherwise we will not see any change at all.**

Reviewer: ```suggestion
removed and installed again on the browser if they are changed. No changes will be reflected without a re-installation.**
```

> re-installed (removed and installed again) on the browser after we change this, otherwise we will not see any change at all.**

Reviewer: what about a short section on how to remove the wallet so that the user can re-install it again if needed? (maybe not in this page)

Author: That's part of "how to use" and "how to add or remove add-ons" on each browser, i think if you know how to add them, remove the wallet should be clear, also we have 4 supported browsers and that section could be a lot of information, let's just assume that the user is an advanced user, because who uses this for development is not a basic user actually
